### PR TITLE
feat(tensorrt_common): add gtest

### PR DESCRIPTION
**Pull request by cyn-liu**
## Description

add gtest to tensorrt_common

## Tests performed

unit test

## Effects on system behavior

There is no effect on the function.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tensorrt_common/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,45 @@ if(NOT (CUDAToolkit_FOUND AND CUDNN_FOUND AND TENSORRT_FOUND))
   return()
 endif()
 
+# Download onnx
```
Comment (mitsudome-r) [inline common/tensorrt_common/CMakeLists.txt:19]: We decided not to include download scripts in CMake. 
Maybe we could skip TestGetInputDims for now.

Comment (cyn-liu) [reply common/tensorrt_common/CMakeLists.txt:19]: To be honest, this package is not really used in autoware.

Comment (mitsudome-r) [reply common/tensorrt_common/CMakeLists.txt:19]: Are you sure about that? The header files seems to be included by the other packages. ([example](https://github.com/autowarefoundation/autoware.universe/blob/0dee51263a607ea19f99f9de34e26668368547a1/perception/tensorrt_yolox/include/tensorrt_yolox/tensorrt_yolox.hpp#L21))

I will discuss this in the next Software WG if we could consider adding sample models in the repository or not for tests.

Comment (cyn-liu) [reply common/tensorrt_common/CMakeLists.txt:19]: > To be honest, this package is not really used in autoware.

@mitsudome-r 
sorry, I didn't make it clear, there are 3 packages in the  perception folder related to tensorrt: `tensorrt_classifier` , `tensorrt_yolo`, `tensorrt_yolox`, where both `tensorrt_classifier`  and `tensorrt_yolox` contain the `tensorrt_common`, but `tensorrt_yolo` does not contain it.  for our real car tests we only used the `tensorrt_yolo`.

**Comment by xmfcx** [comment]
Related:
- https://github.com/autowarefoundation/autoware.universe/issues/7724

**Pull request closed without merging**